File: challenge_questions/challenge_1/isPrime.java

//? Onyx, 2026.
//? https://github.com/kokovt

import java.util.Scanner;

public class isPrime {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Please input a number to check: ");
        long inputNumber = input.nextLong();
        long start = System.currentTimeMillis();
        boolean prime = isPrime.primeTest(inputNumber);
        long elapsedTimeMillis = System.currentTimeMillis()-start;

        System.out.printf("The function took %d milliseconds.%n",elapsedTimeMillis);

        if (prime) {
            System.out.printf("The number %d is prime.", inputNumber);
            input.close();
            return;
        }
        System.out.printf("The number %d is not prime.", inputNumber);
        input.close();
        return;
    }

    public static boolean primeTest(long inputNumber) {
        if (inputNumber <= 1L)
            return false;
        if (inputNumber == 2L)
            return true;
        if (inputNumber % 2 == 0)
            return false;

        long limit = (long) Math.ceil(Math.sqrt(inputNumber));

        for (long i = 2; i < limit; i++) {
            if (inputNumber % i == 0)
                return false;
        }

        return true;
    }
}

File: challenge_questions/challenge_1/isPrime.js
//? Onyx, 2026.
//? https://github.com/kokovt

//? Example algorithm in a way I know.

function isPrime(number) {
    console.log(number);
    // Any number 1 and under is not prime.
    if(number <= 1) return false;

    // Two is prime. This will help with checking evens without false positives.
    if(number == 2) return true;

    // If it is divisible by 2, then it is not prime.
    if(number % 2 == 0) return false;

    // Check all numbers that are less than the square root of the number.
    // The number is prime if no factors are found up to sqrt(n).

    const limit = Math.ceil(Math.sqrt(number));
    for(let i = 2; i <= limit; i++) {
        // console.log(i);
        if(number%i == 0) return false;
    }

    return true;
}

console.log(isPrime(2147483647));
console.log(isPrime(2147483629));
console.log(isPrime(3215031751));
console.log(isPrime(1231551679889));